### DISCUSSION TITLE
Job delete StatusDetails

### DISCUSSION THREAD

**Discussion by hambone01** [fabric8io/kubernetes-client#4592] [General]
I am upgrading **kubernetes-client** from 5.11.2 to 6.2.0. I start/watch/delete Jobs and can't find any documentation on how to properly use the StatusDetails to confirm a Job was deleted once it has finished. Can anyone point me to documentation or maybe an example?

Thank you for your attention.

**Comment by shawkins** [comment]
> Can anyone point me to documentation or maybe an example?

StatusDetails only provide information on what has been marked for deletion.  If you want to know when the job has been deleted you have to correlate the uid in StatusDetails with a delete event or ensure that the corresponding resource no longer exists.  

#4578 is a proposal for a blocking delete operation.  The core part of that change is from https://github.com/fabric8io/kubernetes-client/pull/4578/files#diff-707b35b77f89c083b95f7151bae8eefa6e1f2df7ec1deaa423f8559265b36730R453

That can be generalized as: 

```
    // perform a delete
    List<StatusDetails> deleted = op.delete(); 
    // find the set of resources marked for deletion
    Set<String> uids = deleted.stream().map(StatusDetails::getUid).collect(Collectors.toSet()); 
    // create a future to be informed when they are all gone
    CompletableFuture<List<T>> delete = this
          .informOnCondition(l -> l.stream().map(i -> i.getMetadata().getUid()).noneMatch(uid -> uids.contains(uid)));
```
